packages/foam-core/src/markdown-provider.ts

```diff
@@ -171,12 +171,33 @@ const getTextFromChildren = (root: Node): string => {
 
 const tagsPlugin: ParserPlugin = {
   name: 'tags',
-  onWillVisitTree: (tree, note) => {
-    note.tags = extractHashtags(note.source.text);
+  onDidFindProperties: (props, note, node) => {
+    if (isSome(props.tags)) {
+      const yamlTags = extractTagsFromProp(props.tags);
+      yamlTags.forEach(t => {
+        note.tags.push({
+          label: t,
+          range: astPositionToFoamRange(node.position!),
+        });
+      });
+    }
   },
-  onDidFindProperties: (props, note) => {
-    const yamlTags = extractTagsFromProp(props.tags);
-    yamlTags.forEach(tag => note.tags.add(tag));
+  visit: (node, note) => {
+    if (node.type === 'text') {
+      const tags = extractHashtags((node as any).value);
```

Comment:
Is there a reason for the `as any` here vs `as string`? `any` doesn't feel right given the `extractHashtags` function signature